### PR TITLE
Version history table rendering update for accessibility

### PR DESCRIPTION
**Pull request by agr**
Addresses https://github.com/NuGet/NuGetGallery/issues/8196

Previously, version history table always had 2 hardly visible columns: 1 for signature information and another for deprecation information. When user is not authenticated on the web site, we never show the package signature information, so the cells although exist are always empty. The deprecation information is present, but most of the time packages are not deprecated and empty cells do not have any narrator markup, so are not read (but headers are).

This change moves the `Model.IsCertificatesUIEnabled` check up to the lines that generate column header, causing the certificate column to be completely absent for unauthenticated users. Same is done for `Model.IsPackageDeprecationEnabled` but it is always `true` in our case. Also `aria-label` properties added for empty cells when those are rendered.

### CODE REVIEW

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml [hunk 1]
```diff
@@ -723,7 +729,7 @@
 
                                         <td>
                                             @if (packageStatusSummary == PackageStatusSummary.Listed ||
-                                                 packageStatusSummary == PackageStatusSummary.Unlisted)
+     packageStatusSummary == PackageStatusSummary.Unlisted)
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml:732]: Was this change intentional? I prefer the old styling :)

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml [hunk 2]
```diff
@@ -734,15 +740,22 @@
...
-                                    <td class="package-icon-cell">
-                                        @if (Model.IsPackageDeprecationEnabled && packageVersion.DeprecationStatus != PackageDeprecationStatus.NotDeprecated)
+                                        else
+                                        {
+                                            <td class="package-icon-cell" aria-label="No signature"></td>
+                                        }
+                                    }
+                                    @if (Model.IsPackageDeprecationEnabled)
+                                    {
+                                        if (packageVersion.DeprecationStatus != PackageDeprecationStatus.NotDeprecated)
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackage.cshtml:758]: Nit: consider flipping this condition as the `else` block is much smaller. I don't feel strongly about this though

**Review by loic-sharma — APPROVED** [review]

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml (hunk 2 above)
Comment (cristinamanum) [reply src/NuGetGallery/Views/Packages/DisplayPackage.cshtml:758]: I agree it is easier to read packageVersion.DeprecationStatus == PackageDeprecationStatus.Deprecated than two negations.

**Review by cristinamanum — APPROVED** [review]

File: src/NuGetGallery/Views/Packages/DisplayPackage.cshtml (hunk 1 above)
Comment (agr) [reply src/NuGetGallery/Views/Packages/DisplayPackage.cshtml:732]: VS messes up formatting in this file really bad when I edit it.

**Review by zhhyu — APPROVED** [review]
